flocker/restapi/test/test_infrastructure.py

```diff
@@ -90,6 +211,16 @@ def test_illegalResult(self, logger):
         If the return value of the decorated function cannot be I{JSON} encoded
         then the response generated has the I{INTERNAL SERVER ERROR} code.
         """
+        # Workaround the message validation code in @validateLogging,
+        # otherwise the invalid message being logged throws an exception!
+        # See https://github.com/ClusterHQ/eliot/issues/150
+        def cleanup():
+            for message in logger.messages:
+                if message.get("action_type") == "api:json_request":
+                    if message["json"].get("foo"):
+                        message["json"]["foo"] = None
+        self.addCleanup(cleanup)
```

Comment:
Second traceback only happens because of `validateLogging`, yeah - it does extra validation. In real code the log message will be ... dropped I guess. Until https://github.com/ClusterHQ/eliot/issues/45 is fixed at which point it will actually log some useful error log message. That's on next eliot milestone.